prebid-server/pbs-modules/rules-engine.md

```diff
@@ -0,0 +1,805 @@
+
+![Prebid Server Rules Engine Module](/assets/images/prebid-server/pbs_rules_engine_context.png){:class="pb-xlg-img"}
+
+1. Auction requests come in that trigger the Rules Engine Module.
+1. The module reads the current set of rules from account configuration or by directly fetching them from the service provider.
+1. The results of the optimization decision and the auction are logged via analytics.
+1. Eventually the configuration is updated with refined rules.
+
+The initial goal of this module is to define the runtime capabilities that are required to surgically select bidders. We've defined a technical interface and a runtime behavior. Some uses of the module might be manually configured, but the most valuable use of the system will be in the periodic offline machine learning. The goal there is for Prebid to step out of the way and let members of the ecosystem compete to do the best job.
+
+We envision a time when, like the floors sytem, there are multiple vendors providing optimization services of different kinds.
+
+## Configuration Details
+
+Configuration is covered in several sections:
+
+1. [Module configuration syntax](#module-account-level-configuration) - the JSON used to express which bidders should run when. This is basically a kind of a Domain-Specific Language (DSL) where the 
+1. [Rule Trees](#rule-trees) - groups of rules should be thought of as 'trees' split along dimensions like 'datacenter' and 'channel'. These trees are processed in a particular way.
+1. [Schema Functions](#schema-functions) - which dimensions and comparisons can be used in the configuration. 
+1. [Results Functions](#results-functions) - the specific outcomes once a specific rule is chosen.
+1. [Runtime Processing](#runtime-processing) - how it all comes together.
+
+### Module Account-Level Configuration
+
+These are the full set of attributes that can be present in the account-level configuration:
+
+{: .table .table-bordered .table-striped }
+| Property | Scope | Type | Description | Example |
+|----------|-------|------|-------------|---------|
+| enabled | optional | boolean | Whether the module as a whole is enabled. Allows for easy temporary deactivation. Defaults to true. | true |
+| generateRulesFromBidderConfig | optional | boolean | (PBS-Go only) If true, the module automatically generates rules based on 'geoscope' section of the bidder YAML files to include or exclude that bidder from certain countries. Defaults to true. | false |
+| ruleSets[] | required | array of objects | One or more independent sets of rules. | |
+| ruleSets[]<wbr>.enabled | optional | boolean | Indicates whether this ruleset is active. Defaults to true. Allows easy toggling for troubleshooting. | true |
+| ruleSets[]<wbr>.timestamp | optional | string | String indicates when this ruleset was last updated. Useful in optimization and troubleshooting. Format is ISO 8601. | "20250101T00:00:00" |
```

Comment:
Is this supposed to be on the ruleset or is it a top-level field? I assume the ruleset is correct. Go will need to make an update as we followed the example snippet in the docs that show it as a top-level field but that conflicts with the PRD description. I can update the PRD once confirmed.